src/main/java/net/coderbot/iris/config/IrisConfig.java

```diff
@@ -0,0 +1,125 @@
+package net.coderbot.iris.config;
+
+import net.fabricmc.loader.api.FabricLoader;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+/**
+ * A class dedicated to storing the config values of shaderpacks. Right now it only stores the path to the current shaderpack
+ */
+public class IrisConfig {
+    private Path shaderpath;
+    private Path propertiesPath;
+    /**
+     * Represents if the current loaded shaderpack is in internal one or not
+     */
+    private boolean isInternal = false;
+
+    public IrisConfig() {
+        propertiesPath = FabricLoader.getInstance().getConfigDir().resolve("iris.properties");
+        try {
+            Files.createDirectories(FabricLoader.getInstance().getGameDir().resolve("shaderpacks"));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * This sets the default shaderpack name for the shader properties to set.
+     * This value will only be used if the iris.properties file is not found
+     * @param shaderPackName the file name of the pack
+     * @return the shaderproperties with the set value
+     */
+    public IrisConfig setDefaultPack(String shaderPackName){
+        this.shaderpath = FabricLoader.getInstance().getGameDir().resolve("shaderpacks").resolve(shaderPackName);
+        return this;
+    }
```

Comment:
I think that the internal shaders should always be the default, this method probably doesn't need to be exposed publicly.